tracehandler/tracehandler.go

```diff
@@ -77,29 +76,45 @@ type traceHandler struct {
 }
 
 // newTraceHandler creates a new traceHandler
-func newTraceHandler(rep reporter.TraceReporter, traceProcessor TraceProcessor,
-	intervals Times, cacheSize uint32) (*traceHandler, error) {
-	bpfTraceCache, err := lru.New[host.TraceHash, libpf.TraceHash](
+func newTraceHandler(ctx context.Context, rep reporter.TraceReporter,
+	traceProcessor TraceProcessor, intervals Times, cacheSize uint32) (*traceHandler, error) {
+	traceCache, err := lru.NewSynced[host.TraceHash, *libpf.Trace](
 		cacheSize, func(k host.TraceHash) uint32 { return uint32(k) })
 	if err != nil {
 		return nil, err
 	}
+	// Do not hold elements indefinitely in the cache.
+	traceCache.SetLifetime(traceCacheLifetime)
 
-	umTraceCache, err := lru.New[libpf.TraceHash, libpf.Void](
-		cacheSize, libpf.TraceHash.Hash32)
-	if err != nil {
-		return nil, err
-	}
+	var wg sync.WaitGroup
+	wg.Add(1)
+
+	go func() {
+		wg.Done()
+		jitter := 0.2
+		ticker := time.NewTicker(libpf.AddJitter(traceCacheLifetime, jitter))
+		defer ticker.Stop()
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				traceCache.PurgeExpired()
+				ticker.Reset(libpf.AddJitter(traceCacheLifetime, jitter))
+			}
+		}
+	}()
 
-	t := &traceHandler{
+	// Wait to make sure the purge routine did start.
+	wg.Wait()
```

Comment:
Does it make sense to wait here? It doesn't harm if the go routine starts after return.

Reply:
The waitgroup pattern is just used to make sure the Go routine starts. This pattern is used also in other places in the code base. Waiting here should not increase the start up time in a noticeable way. But as we can not control the Go scheduler, we make sure this way, the clean up/purge routine is working properly and scheduled.